packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -284,3 +323,26 @@ def _get_python_type_as_str(type: Field.Kind.ValueType, is_array: bool) -> str:
     if is_array:
         return f"List[{python_type.__name__}]"
     return python_type.__name__
+
+
+def _get_enum_type(parameter: Any, enum_values_by_type_name: Dict[str, Dict[str, Any]]) -> Any:
+    if parameter.type == FieldDescriptorProto.TYPE_ENUM:
+        loaded_enum_values = json.loads(parameter.annotations["ni/enum.values"])
+        enum_values = dict((key, value) for key, value in loaded_enum_values.items())
+
+        for existing_enum_name, existing_enum_values in enum_values_by_type_name.items():
+            if existing_enum_values == enum_values:
+                return Enum(existing_enum_name, existing_enum_values)
+
+        new_enum_type_name = _get_enum_class_name(parameter.name)
+        enum_values_by_type_name[new_enum_type_name] = enum_values
+        return Enum(new_enum_type_name, enum_values)
+    return None
+
+
+def _get_enum_class_name(name: str) -> str:
+    class_name = name.title().replace(" ", "")
+    invalid_chars = "`~!@#$%^&*()-+={}[]\|:;',<>.?/ \n_"
+    pattern = f"[{re.escape(invalid_chars)}]"
+    class_name = re.sub(pattern, "", class_name)
+    return class_name + "Enum"
```

Comment:
See comments on 2nd copy

Reply:
Refactored this method based on your previous comments. Closing this and continuing the conversation in this [thread](https://github.com/ni/measurement-plugin-python/pull/880#discussion_r1755491884).